loader/src/phase2.rs

```diff
@@ -11,6 +11,36 @@ pub fn phase_2(cfg: &mut BootConfig) {
     let mut process_offset = cfg.sram_start as usize + cfg.sram_size - cfg.init_size;
     println!("\n\nPhase2: Processess start out @ {:08x}", process_offset);
 
+    // Construct an environment block. This will be used by processes
+    // to access environment variables and system parameters. This
+    // is hardcoded for now, and can be expanded later.
+    #[rustfmt::skip]
+    let mut env = [
+        0x41, 0x70, 0x70, 0x50, // 'AppP' indicating application parameters
+        0x08, 0x00, 0x00, 0x00, // Size of AppP tag contents
+        0xb2, 0x00, 0x00, 0x00, // Size of entire AppP block including all tags
+        0x02, 0x00, 0x00, 0x00, // Number of tags present
+        0x45, 0x6e, 0x76, 0x42, // 'EnvB' indicating an environment block
+        0x9a, 0x00, 0x00, 0x00, // Number of bytes that follows for the environment block
+        0x01, 0x00, // Number of environment variables
+        0x14, 0x00, // Length of name of first variable
+        // Name of first variable 'ROOT_FILESYSTEM_HASH':
+        0x52, 0x4f, 0x4f, 0x54, 0x5f, 0x46, 0x49, 0x4c, 0x45, 0x53, 0x59, 0x53, 0x54, 0x45, 0x4d,
+        0x5f, 0x48, 0x41, 0x53, 0x48,
+        // Length of the contents of the first variable
+        0x80, 0x00,
+        // Root filesystem hash contents begin here:
+        0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30,
```

Comment:
so i just jam the hash into here, as ASCII hex digits correct?

Reply:
Yep. As long as the length doesn't change, it'll be visible in all programs as an environment variable.

Reply:
yah i see the warning now. thanks!